api/src/main/java/com/velocitypowered/api/proxy/Player.java

```diff
@@ -58,5 +55,16 @@ default void sendMessage(@NonNull Component component) {
      */
     ConnectionRequestBuilder createConnectionRequest(@NonNull ServerInfo info);
 
+    /**
+     * Sets a header and footer to the player
+     * @param header component with header
+     * @param footer component with footer
+     */
+    void setHeaderAndFooter(@NonNull Component header, @NonNull Component footer);
```

Comment:
A method to conveniently clear the header and footer would be nice.

Reply:
The `@NonNull` annotations can be safely dropped, since the Checker Framework assumes `@NonNull` by default.